### ISSUE TITLE
Make Operate/Tasklist snapshot artifacts public

### ISSUE THREAD

**Issue by cmur2** [camunda/camunda#19035]
## Description

When analyzing the [current status of Maven Artifact Distribution](https://docs.google.com/document/d/124XowQdzBB6nRISRmgknW-RlI5v5JolJL7XTpjd-348/edit#heading=h.7slsh1rrddjw) in the C8 monorepo we found an inconsistency: while the source code of all components is public, snapshot artifacts of Operate/Tasklist are not publicly accessible on https://artifacts.camunda.com/ unlike Zeebe due to historic reasons. This can be an obstacle for Camunda and external contributors.

We decided to overcome this by creating a new pair of repositories (name TBD) on https://artifacts.camunda.com/ that should contain all future releases of all components in this C8 monorepo. This needs infrastructure help from Infra team.

This also needs to be done for all future components being migrated to this C8 monorepo.

**Comment by maxdanilov** [comment]
@cmur2 is this still relevant for the current version of Camunda 8?

**Comment by cmur2** [comment]
> [@cmur2](https://github.com/cmur2) is this still relevant for the current version of Camunda 8?

I don't recall anyone working on this and just checked: this inconsistency is still in place indeed.

**Comment by maxdanilov** [comment]
But don't we plan in the future to stop shipping separate Operate/Tasklist components to Artifactory and all will be in the big JAR (if not already)?

**Comment by cmur2** [comment]
> But don't we plan in the future to stop shipping separate Operate/Tasklist components to Artifactory and all will be in the big JAR (if not already)?

I had a very [fruitful discussion](https://camunda.slack.com/archives/C08F5RD5X8B/p1749642821631049) with Thorben on that matter and have now a clearer picture of what we need, for engineers to have better UX when working in the monorepo. I'll update the ticket description accordingly (motivation, possible solutions, implementation hints).

TL;DR: As far as I understand it we'll continue shipping JAR files for each Maven module in the monorepo (Zeebe, Operate, Tasklist, QA stuff, etc) as that is how Maven works. When building e.g. the `camunda/camunda` Docker image using the "single" JAR, we build [an "uber" JAR](https://maven.apache.org/plugins/maven-shade-plugin/) on the fly to use in that - but those uber JARs are not typically uploaded to artifact hosters since they mix our and 3rd party code etc. -> ticket is still relevant and decoupled from the "single" JAR question.

**Comment by cmur2** [comment]
PR for reviewer:

* https://github.com/camunda/camunda/pull/35376

**Comment by cmur2** [comment]
The change from https://github.com/camunda/camunda/pull/35376 is now active on `main`, `stable/8.7` and `stable/8.6` and was announced to affected Camundi to gather their feedback.

**Comment by cmur2** [comment]
Closing as Camundi reported the change to be effective
